akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -628,6 +821,15 @@ final class ClusterReceptionist(pubSubMediator: ActorRef, settings: ClusterRecep
   val virtualNodesFactor = 10
   var consistentHash: ConsistentHash[Address] = ConsistentHash(nodes, virtualNodesFactor)
 
+  var clientInteractions = HashMap.empty[ActorRef, DeadlineFailureDetector]
+  var clientsPublished = HashSet.empty[ActorRef]
+
+  context.system.scheduler.schedule(
```

Comment:
Is this a requirement when scheduling against an actor ref? I had assumed that it'd just be watched and self-cancel (perhaps a bad assumption)... or is it just a good pattern of behaviour that you seek...?